docs/configuration/git-providers/bitbucket.md

```diff
@@ -0,0 +1,35 @@
+# Bitbucket
+
+Since Bitbucket uses OAuth, the first step is to register a consumer for Codacy on Bitbucket side.
+
+## Create a consumer
+OAuth needs a key and secret, together these are known as an OAuth consumer. You can create a consumer on any existing individual or team account. To create a consumer, do the following:
+
+  1. From your avatar in the bottom left, click Bitbucket settings.
+  1. Click *"OAuth"* from the left navigation and click the *"Add consumer"* button.  
+     1. Set the name of the consumer (e.g. *"Codacy"*).
+     1. Set the callback URL as *"http://<codacy_url>:<codacy_port>/login/Bitbucket?codacy_skip_ga=1"*.
+     1. Tick the *"This is a private consumer"* checkbox:
+        ![Bitbucket Consumer Configuration](./images/bitbucket-consumer-configuration.png)
+     1. Add the following permissions: 
+     ![Bitbucket Consumer Permissions](./images/bitbucket-consumer-permissions.png)
+     1. Click Save. 
+        The system generates a key and a secret for you.
+     1. Toggle the consumer name to see the generated Key and Secret value for your consumer.
+     ![Bitbucket Consumer Key and Secret](./images/bitbucket-consumer-key-and-secret.png)
+
+## Enable Bitbucket and set the key and the secret
+Set your configuration values for Bitbucket on the `values.yaml` file.
+
+1. Enable Bitbucket by setting `enable` to `"true"`.
+1. Copy the key from Bitbucket and set it in the `key` field.
+1. Copy the secret from Bitbucket and set it in the `secret`field.
+
+Example:
+```yaml
+global:
+  bitbucket:
+    enabled: "true"
+    key: "12345232123"
+    secret: "54321abcxdawj4bfsdh3"
+```
```

Comment:
We also need to run a command to update the settings on the cluster, something like:

```
helm upgrade (...options used to install Codacy...) \
              --values values-production.yaml
```

Reply:
should we repeat this command for every provider instructions?

Reply:
I will make sure this is consistent for all the Git providers when both the Inquisition and Jonah's finish their current work. So, let's ensure the instructions are correct for Bitbucket Cloud in this PR and I'll take care of the other providers.